exoscale/datasource_exoscale_compute_ipaddress.go

```diff
@@ -0,0 +1,154 @@
+package exoscale
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/exoscale/egoscale"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceComputeIPAddress() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:        schema.TypeString,
+				Description: "Name of the zone",
+				Required:    true,
+			},
+			"description": {
+				Type:          schema.TypeString,
+				Description:   "Description of the IP",
+				Optional:      true,
+				ConflictsWith: []string{"ip_address", "id", "tags"},
+			},
+			"ip_address": {
+				Type:          schema.TypeString,
+				Description:   "IP Address",
+				Optional:      true,
+				ConflictsWith: []string{"description", "id", "tags"},
+			},
+			"id": {
+				Type:          schema.TypeString,
+				Description:   "ID of the IP Address",
+				Optional:      true,
+				ConflictsWith: []string{"description", "ip_address", "tags"},
+			},
+			"tags": {
+				Type:          schema.TypeMap,
+				Description:   "Map of tags (key: value)",
+				Optional:      true,
+				ConflictsWith: []string{"description", "ip_address", "id"},
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+
+		Read: datasourceComputeIPAddressRead,
+	}
+}
+
+func datasourceComputeIPAddressRead(d *schema.ResourceData, meta interface{}) error {
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutCreate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	zoneName := d.Get("zone").(string)
+	zone, err := getZoneByName(ctx, client, zoneName)
+	if err != nil {
+		return err
+	}
+
+	id := d.Get("id").(string)
+
+	var uuid *egoscale.UUID
+	if id != "" {
+		uuid = egoscale.MustParseUUID(id)
+	}
+
+	resp, err := client.RequestWithContext(ctx,
+		egoscale.ListPublicIPAddresses{
+			ZoneID:    zone.ID,
+			ID:        uuid,
+			IPAddress: net.ParseIP(d.Get("ip_address").(string)),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	ips := resp.(*egoscale.ListPublicIPAddressesResponse).PublicIPAddress
+
+	t, ok := d.GetOk("tags")
```

Comment:
Please use a more relevant variable name when it's not used immediately after its assignment:

```suggestion
	t, byTags := d.GetOk("tags")
```